vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oauth2/ClientAuthenticationService.kt

```diff
@@ -52,8 +52,8 @@ class ClientAuthenticationService(
             ).getOrElse {
                 throw InvalidClient("could not parse client attestation", it)
             }
-            if (!verifyJwsObject(clientAttestationJwt)) {
-                throw InvalidClient("client attestation JWT not verified")
+            verifyJwsObject(clientAttestationJwt).getOrElse {
+                throw InvalidClient("client attestation JWT not verified. $it")
```

Comment:
Need to pass `it` as the second parameter to the `InvalidClient` constructor, so the cause is set correctly. This applies to several lines.